src/commands/start_project.rs

```diff
@@ -0,0 +1,45 @@
+async fn template_contents(project_name: &str) -> (String, String) {
+    let ftd = format!("-- import: fpm\n\n-- fpm.package: {}", project_name);
+    let index = "-- ftd.text: Hello world".to_string();
+
+    (ftd, index)
+}
+
+async fn write_file(file_name: &str, dir: &str, content: &str) -> fpm::Result<()> {
+    use tokio::io::AsyncWriteExt;
+    let file_path = format!("{}/{}", dir, file_name);
+    let mut fp = tokio::fs::File::create(file_path).await?;
+    fp.write_all(content.as_bytes()).await?;
+    Ok(())
+}
+
+pub async fn start_project(name: Option<&str>, path: Option<&str>) -> fpm::Result<()> {
+    let base_path = std::env::current_dir()?.to_str().unwrap().to_string();
+    // Not using config for base path as it requires manifest or FPM.ftd file for building and will throw error
+    // and since this command should work from anywhere within the system
+    // so we dont need to rely on config for using it
+
+    // name is a required field so it will always be some defined string (cant be None)
+    // path is an optional field and has a default value "." for current working directory (also cant be None)
+    let (name, path) = (name.unwrap(), path.unwrap());
```

Comment:
This is not right, you should only use unwrap() if it something you are reasonably sure is impossible. If you need some data, do not accept `Option<>`, change the signature to `start_project(name: &str, path: &str)`.

Acception Option, only if it is valid. Here by doing an unwrap() you are saying, "hey, my program will crash if this parameter is not passed". In which case why are you saying something is optional.

Reply:
Seems you are calling start project with optional stuff because that is what that argument returns. In which case keep the function signature as Option, but instead of `.unwrap()`, do a match, and if any requirement argument is missing, print an error message, and return from the function. Do not crash. `.unwrap()` means crash, and crash means bug. If user did a mistake, print a user friendly error message.

Reply:
Seems you are calling start project with optional stuff because that is what that argument returns. In which case keep the function signature as Option, but instead of `.unwrap()`, do a match, and if any requirement argument is missing, print an error message, and return from the function. Do not crash. `.unwrap()` means crash, and crash means bug. If user did a mistake, print a user friendly error message.